client/src/App.js

```diff
@@ -1,29 +1,60 @@
-import { useEffect, useState } from 'react';
-import logo from './logo.svg';
 import './App.css';
+import Navbar from './components/Navbar';
+import Home from './pages/Home';
+import SearchResult from './pages/SearchResult';
+import Favorites from './pages/Favorites';
+import backgroundImage from '../src/pages/serge-le-strat-rS4OSc9yhSo-unsplash.jpg';
+import { Route, Routes } from 'react-router-dom';
+import { useState } from 'react';
+import { useNavigate } from 'react-router-dom';
 
 function App() {
-  const [serverData, setServerData] = useState("");
+  // console.log(window.location);
+  const [search, setSearch] = useState('');
+  const [recipes, setRecipes] = useState([]);
+  const APP_ID = process.env.REACT_APP_ID;
+  const APP_KEY = process.env.REACT_APP_API_KEY;
+  const RECIPE_URL = `https://api.edamam.com/api/recipes/v2?type=public&app_id=${APP_ID}&app_key=${APP_KEY}&q=${search}`;
+  const navigate = useNavigate();
 
-  useEffect(() => {
-    async function getServerData() {
-      const resp = await fetch('/api/hello');
-      const data = await resp.json();
+  const handleChange = (e) => {
+    setSearch(e.target.value);
+  };
 
-      console.log('Data from server:', data);
-
-      setServerData(data.message);
+  const handleSubmit = async (e) => {
+    e.preventDefault();
+    try {
+      const response = await fetch(RECIPE_URL);
+      if (!response.ok) {
+        throw new Error(400, 'recipe not found');
```

Comment:
Error doesn't have a constructor like this. It only takes a single parameter, a string message. And it may not be a "not found" error. Including the response.status or `await response.text()` in the error message would be helpful.